### PR TITLE
Parameter node wrapper docs

### PR DESCRIPTION
**Pull request by Connor-Bowley**
Adds a small new feature for setting Decimals and SingleStep

Fully revamps documentation for the parameterNodeWrapper.

### CODE REVIEW

File: Docs/developer_guide/parameter_nodes/advanced/custom_classes.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+:::
+
+## Using custom classes in a parameterNodeWrapper
+
+This section is meant as a guide to make new types available for use in the `parameterNodeWrapper` and `parameterPack`s. The serialization section must always be done, but the GUI sections are only needed if you want to support GUI connection or creation with your type. New types should be able to be added in extensions (but this is untested).
+
+### Serialization into the underlying vtkMRMLScriptedModuleNode
+
+<details>
+  <summary>Click to expand</summary>
```
Comment (allemangD) [inline Docs/developer_guide/parameter_nodes/advanced/custom_classes.md:16]: I'd prefer these not be collapsible; none of the other content collapses like this and it is a bit more difficult to read.

**Review by allemangD — APPROVED** [review]
Wow! I like the new docs - great work. One nitpick but I've read through and it seems comprehensive.

**Review by sjh26 — DISMISSED** [review]
lgtm

File: Docs/developer_guide/parameter_nodes/gui_connection.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+There are two ways to declare which GUI widget elements connect to which parameters.
+
+## Connecting widgets from a .ui file
+
+Qt Designer has the ability to set Dynamic Properties. These can be used to inform the parameter node wrapper infrastructure which parameters to connect to which widgets. This is the preferred way of connection from the `.ui` file of a scripted module to its parameter node wrapper.
+
+Simply set a dynamic string property named "SlicerParameterName" to the parameter name it should bind with.
+
+![Qt Dynamic Property](gui_connection-QtDynamicProperty.png)
```
Comment (jcfr) [inline Docs/developer_guide/parameter_nodes/gui_connection.md:15]:
```suggestion
![Qt Dynamic Property](https://github.com/Slicer/Slicer/releases/download/docs-resources/parameter_node_gui_connection_QtDynamicProperty.png)
```

Now that I uploaded the image to https://github.com/Slicer/Slicer/releases/tag/docs-resources, consider removing it from this pull-request.

File: Docs/developer_guide/parameter_nodes/advanced/custom_classes.md (hunk 1 above)
Comment (Connor-Bowley) [reply Docs/developer_guide/parameter_nodes/advanced/custom_classes.md]: Removed the collapsibles.

**Review by allemangD — APPROVED** [review]

**Pull request merged**